### ISSUE TITLE
If VM is manually deleted outside of terraform the provider errors

### ISSUE THREAD

**Issue by rmb938** [terra-farm/terraform-provider-xenorchestra#79]
If I let terraform create the VM via a `terraform apply` then manually delete the VM in Xen Orchestra the provider errors instead of trying to re-create the VM

```
terraform {
  required_providers {
    xenorchestra = {
      source = "terra-farm/xenorchestra"
      version = "0.7.1"
    }
  }
}

provider "xenorchestra" {
  # Must be ws or wss
  url = "some url here"
  username = "admin@admin.net"
  password = "admin"
}


data "xenorchestra_template" "template" {
  name_label = "CentOS 7"
}

data "xenorchestra_network" "net" {
  name_label = "Pool-wide network associated with eth0"
}

resource "xenorchestra_cloud_config" "bar" {
  name = "cloud config name"
  template = <<EOF
#cloud-init

EOF
}

resource "xenorchestra_vm" "bar" {
  memory_max = 1073733632
  cpus  = 1
  cloud_config = xenorchestra_cloud_config.bar.name
  name_label = "Name"
  name_description = "description"
  template = data.xenorchestra_template.template.id
  network {
    network_id = data.xenorchestra_network.net.id
  }

  disk {
    sr_id = "97d30a97-1c42-a85e-9921-67d3f4c68cc3"
    name_label = "Name"
    size = 32212254720
  }
}

```

Error:

```
 terraform apply
data.xenorchestra_template.template: Refreshing state... [id=OpaqueRef:57c217c2-06dd-4595-b832-62766cd24fe8]
data.xenorchestra_network.net: Refreshing state... [id=2d2f8511-39f6-9417-9f72-a334900f195a]
xenorchestra_cloud_config.bar: Refreshing state... [id=62a88f88-cf0b-4c1e-9fe0-4601a981b599]
xenorchestra_vm.bar: Refreshing state... [id=df9c37ab-d008-9d35-df1c-6e8b2bf368e6]

Error: Could not find client.Vm with query: {Type: Id:df9c37ab-d008-9d35-df1c-6e8b2bf368e6 NameDescription: NameLabel: CPUs:{Number:0} Memory:{Dynamic:[] Static:[] Size:0} PowerState: VIFs:[] VirtualizationMode: PoolId: Template: AutoPoweron:false HA: CloudConfig: ResourceSet:}

```

**Comment by ddelnano** [comment]
@rmb938 thanks for bringing this up.

This is a flaw in the other resources of the provider as well. I've started a fix for the VM resource in #80.

**Comment by ddelnano** [comment]
This is fixed in v0.7.2
